### PR TITLE
WIP: Sketch for tests of @daml/ledger streaming methods

### PR DESCRIPTION
**Pull request by hurryabit**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/5286)

### CODE REVIEW

File: language-support/ts/daml-ledger/index.ts [hunk 1]
```diff
@@ -493,10 +493,10 @@ class Ledger {
     let haveSeenEvents = false;
     let state = init;
     const emitter = new EventEmitter();
-    ws.onopen = () => {
+    ws.addEventListener('open', () => {
```
Comment (robin-da) [inline language-support/ts/daml-ledger/index.ts]: why did you change these?

File: language-support/ts/daml-ledger/BUILD.bazel [hunk 2]
```diff
@@ -6,22 +6,20 @@ load("@build_bazel_rules_nodejs//:index.bzl", "pkg_npm")
...
     visibility = ["//visibility:public"],
     deps = [
         "//language-support/ts/daml-types",
-        "@language_support_ts_deps//@mojotech/json-type-validation",
-        "@language_support_ts_deps//@types/ws",
-        "@language_support_ts_deps//cross-fetch",
-        "@language_support_ts_deps//events",
-        "@language_support_ts_deps//isomorphic-ws",
-        "@language_support_ts_deps//ws",
+        "@language_support_ts_deps//:node_modules",
```
Comment (aherrmann-da) [inline language-support/ts/daml-ledger/BUILD.bazel]: Why do you need to depend on all of `node_modules`?

File: language-support/ts/daml-ledger/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  serverOpen(): void;
+  serverSend(message: Message): void;
+  serverClose(event: {code: number; reason: string}): void;
+}
+
+let mockInstance = undefined as unknown as MockWebSocket;
+const mockConstructor = jest.fn();
+const mockSend = jest.fn();
+
+jest.mock('isomorphic-ws', () => class {
```
Comment (aherrmann-da) [inline language-support/ts/daml-ledger/index.test.ts]: Oh, so you can mock node modules inline after all. [The docs](https://jestjs.io/docs/en/manual-mocks) only seem to talk about the `__mocks__` way to do this.

File: language-support/ts/daml-ledger/index.ts (hunk 1 above)
Comment (hurryabit) [reply language-support/ts/daml-ledger/index.ts]: That makes mocking easier since overriding a method is easier than overriding a field with magic behavior.

File: language-support/ts/daml-ledger/BUILD.bazel (hunk 2 above)
Comment (hurryabit) [reply language-support/ts/daml-ledger/BUILD.bazel]: That's what we do in `@daml/react` as well. I pretty much copied from there. I'll try to make it more restrictive here.

File: language-support/ts/daml-ledger/index.test.ts (hunk 3 above)
Comment (hurryabit) [reply language-support/ts/daml-ledger/index.test.ts]: Yeah, that's what we do in `@daml/react` too: https://github.com/digital-asset/daml/blob/7185608f876372df776c5e74dd1e5274cb4e518e/language-support/ts/daml-react/index.test.ts#L21

File: language-support/ts/daml-ledger/BUILD.bazel (hunk 2 above)
Comment (hurryabit) [reply language-support/ts/daml-ledger/BUILD.bazel]: Keeping the old list and only adding `@language_support_ts_deps//@types/jest` does the job too. Doing that now.

Comment (aherrmann) [reply language-support/ts/daml-ledger/BUILD.bazel]: Thanks!

**Comment by hurryabit** [comment]
This sketch has been implemented in https://github.com/digital-asset/daml/pull/5180.
